custom_components/waste_collection_schedule/waste_collection_schedule/source/dechets_gpseo_fr.py

```diff
@@ -0,0 +1,164 @@
+            'emballages': "Emballages",
+            'encombrants': "Encombrants",
+            'dechets-verts': "Déchets Verts",
+            'verres': "Verres"
+        }
+
+        # List to hold collection entries
+        entries = []
+
+        # Loop through each waste type and extract the relevant information
+        for waste_class, waste_name in waste_types.items():
+            block = soup.find('div', class_=f'wrapper-block {waste_class}')
+            
+            if block:
+                print(f"Found block for: {waste_name}")
+                # Look for the date-event field within the block
+                date_block = block.find('div', class_='field--name-field-date-event')
+                if date_block:
+                    date_items = date_block.find_all('div', class_='field__item')
+                    for item in date_items:
+                        date_str = item.text.strip()
+                        date = self._parse_date(date_str)
+                        if date:
+                            entries.append(Collection(date, waste_name, icon=ICON_MAP.get(waste_class)))
+                else:
+                    print(f"No date-event field found within block for {waste_name}")
+            else:
+                print(f"No block found for: {waste_name}")
+        
+        print(f"Entries found: {len(entries)}")
+        return entries
+
+    def _parse_date(self, date_str):
+        # Map French days of the week to integers
+        days_of_week = {
+            "lundi": 0,
+            "mardi": 1,
+            "mercredi": 2,
+            "jeudi": 3,
+            "vendredi": 4,
+            "samedi": 5,
+            "dimanche": 6
+        }
+        
+        # Extract time of day if specified
+        if "matin" in date_str:
+            time_of_day = "morning"
+        elif "après-midi" in date_str:
+            time_of_day = "afternoon"
+        else:
+            time_of_day = "night before"  # Default assumption
+        
+        # Extract the day of the week
+        for day in days_of_week.keys():
+            if day in date_str:
+                today = datetime.now()
+                current_weekday = today.weekday()
+                target_weekday = days_of_week[day]
+
+                # Calculate the next occurrence of the target weekday
+                days_ahead = target_weekday - current_weekday
+                if days_ahead <= 0:  # Target day already passed this week
+                    days_ahead += 7
+                collection_date = today + timedelta(days=days_ahead)
+                
+                # Adjust the date based on the time of day
+                if time_of_day == "night before":
+                    collection_date -= timedelta(days=1)
+
+                return collection_date.date()
+        
+        # Try parsing dates like "12 septembre"
+        try:
+            return datetime.strptime(date_str, "%d %B").replace(year=datetime.now().year).date()
```

Comment:
This cannot parse 12 septembre as it can only parse English dates

so you probably want to add soemthing like this at the top

```py
FRENCH_TO_ENGLISH_MONTHS = {
    "janvier": "January",
    "février": "February",
    "mars": "March",
    "avril": "April",
    "mai": "May",
    "juin": "June",
    "juillet": "July",
    "août": "August",
    "septembre": "September",
    "octobre": "October",
    "novembre": "November",
    "décembre": "December"
}
```

and this bevore trying to parse the date:
```py
        for french_month, english_month in FRENCH_TO_ENGLISH_MONTHS.items():
            date_str = date_str.replace(french_month, english_month)
```